ksql-engine/src/main/java/io/confluent/ksql/KsqlEngine.java

```diff
@@ -312,6 +312,45 @@ public QueryMetadata getQueryExecutionPlan(final Query query, final KsqlConfig k
     }
   }
 
+  public List<Pair<String, Statement>> parseStatements(
+      final String queriesString,
+      final MetaStore tempMetaStore
+  ) {
+    try {
+      final MetaStore tempMetaStoreForParser = tempMetaStore.clone();
+      final KsqlParser ksqlParser = new KsqlParser();
+
+      final List<SqlBaseParser.SingleStatementContext> parsedStatements
+          = ksqlParser.getStatements(queriesString);
+      final List<Pair<String, Statement>> queryList = new ArrayList<>();
+
+      for (SqlBaseParser.SingleStatementContext singleStatementContext : parsedStatements) {
+        final Pair<Statement, DataSourceExtractor> statementInfo = ksqlParser.prepareStatement(
+            singleStatementContext,
+            tempMetaStoreForParser
+        );
+        final String statementString = getStatementString(singleStatementContext);
+        Statement statement = statementInfo.getLeft();
```

Comment:
i'd extract L333 to L339 into a method, i.e., 
```
private Statement getStatement(final Pair<Statement, DataSourceExtractor> statementInfo) {
    final Statement statement = statementInfo.getLeft();
    if (StatementRewriteForStruct.requiresRewrite(statement)) {
               return new StatementRewriteForStruct(
               statement,
               statementInfo.getRight())
               .rewriteForStruct();
        }
   return statement;
}
```

Reply:
Although it would be a very tiny method, I moved it to a new one.

Reply:
Nothing wrong with tiny methods. Big methods on the other hand are very hard to grok.

Reply:
nit: final.